pkg/cloud/azure/services/networkinterfaces/networkinterfaces.go

```diff
@@ -37,7 +39,7 @@ type Spec struct {
 	StaticIPAddress          string
 	PublicLoadBalancerName   string
 	InternalLoadBalancerName string
-	NatRule                  int
```

Comment:
So bastion VM is just a flavour of created cluster, some users may not want additional vm. So please add support for both scenarios